test/parallel/test-http2-session-settings.js

```diff
@@ -8,110 +8,131 @@ const h2 = require('http2');
 
 const server = h2.createServer();
 
-server.on('stream', common.mustCall(onStream));
-
-function assertSettings(settings) {
-  assert.strictEqual(typeof settings, 'object');
-  assert.strictEqual(typeof settings.headerTableSize, 'number');
-  assert.strictEqual(typeof settings.enablePush, 'boolean');
-  assert.strictEqual(typeof settings.initialWindowSize, 'number');
-  assert.strictEqual(typeof settings.maxFrameSize, 'number');
-  assert.strictEqual(typeof settings.maxConcurrentStreams, 'number');
-  assert.strictEqual(typeof settings.maxHeaderListSize, 'number');
-}
-
-function onStream(stream, headers, flags) {
-
-  const localSettings = stream.session.localSettings;
-  const remoteSettings = stream.session.remoteSettings;
-  assertSettings(localSettings);
-  assertSettings(remoteSettings);
-
-  // Test that stored settings are returned when called for second time
-  assert.strictEqual(stream.session.localSettings, localSettings);
-  assert.strictEqual(stream.session.remoteSettings, remoteSettings);
-
-  stream.respond({
-    'content-type': 'text/html',
-    ':status': 200
-  });
-  stream.end('hello world');
-}
-
-server.listen(0);
-
-server.on('listening', common.mustCall(() => {
-
-  const client = h2.connect(`http://localhost:${server.address().port}`, {
-    settings: {
-      enablePush: false,
-      initialWindowSize: 123456
-    }
-  });
-
-  client.on('localSettings', common.mustCall((settings) => {
-    assert(settings);
-    assert.strictEqual(settings.enablePush, false);
-    assert.strictEqual(settings.initialWindowSize, 123456);
-    assert.strictEqual(settings.maxFrameSize, 16384);
-  }, 2));
-  client.on('remoteSettings', common.mustCall((settings) => {
-    assert(settings);
-  }));
-
-  const headers = { ':path': '/' };
-
-  const req = client.request(headers);
-
-  req.on('connect', common.mustCall(() => {
-    // pendingSettingsAck will be true if a SETTINGS frame
-    // has been sent but we are still waiting for an acknowledgement
-    assert(client.pendingSettingsAck);
-  }));
-
-  // State will only be valid after connect event is emitted
-  req.on('ready', common.mustCall(() => {
-    assert.doesNotThrow(() => {
-      client.settings({
-        maxHeaderListSize: 1
-      });
+server.on(
```

Comment:
I'm not sure I'm following what's going on with the changes in this file. Is there a reason everything needs to be scoped within the stream handler? This looks like a lot of churn and I'm having trouble following why.

Reply:
I originally started working on this file and formatted the code using Prettier.
I created `test-http2-client-settings-errors.js` after coming across the template used by other tests which checked for errors.
I've scoped method `assertSettings` within stream handler, as it's only used within that block.

Reply:
My concern is just that it's a bit easier to follow the file history (`git blame`) when changes aren't made just for the sake of changing code style, unless the clarity is a significant problem. That said, this is a nit so if no one else cares then this is fine. 👍

Reply:
I agree that code change shouldn't be made just for changing code style.
I added changed in `test-http2-session-settings.js` as the formatting was already done. I'll remove it from the commit if more people raise it in the reviews.
Btw, I created an issue at #16115 to check if we can move to Javascript formatter like Prettier.